src/pkcs11/mechanism.c

```diff
@@ -1315,6 +1315,16 @@ sc_pkcs11_new_fw_mechanism(CK_MECHANISM_TYPE mech,
 	return mt;
 }
 
+void sc_pkcs11_free_mechanism(sc_pkcs11_mechanism_type_t *mt)
+{
+	if (!mt)
+		return;
+	if (mt->free_mech_data)
+		mt->free_mech_data(mt->mech_data);
+	free(mt);
+	mt = NULL;
```

Comment:
`mt = NULL` will most likely be optimized out, because the code doesn't do anything with the new value

Reply:
Removed